### PR TITLE
fix: move fees to order factory

### PR DESCRIPTION
**Pull request by berteotti**
- Update tests
- Update README

### CODE REVIEW

File: packages/contracts/src/DCAOrder.sol [hunk 1]
```diff
@@ -59,24 +57,22 @@ contract DCAOrder is IConditionalOrder, EIP1271Verifier, IDCAOrder {
...
   /// @param _interval The frequency interval of the DCA order in hours.
   /// @param _settlementContract The settlement contract address.
-  /// @param _fee The protocol fee amount
   function initialize(
     address _owner,
     address _receiver,
     address _sellToken,
     address _buyToken,
-    uint256 _principal,
+    uint256 _totalAmount,
```
Comment (NelsonGaldeman) [inline packages/contracts/src/DCAOrder.sol]: I might be nitpicky here but can we just call it `amount`?

File: packages/contracts/src/OrderFactory.sol [hunk 2]
```diff
@@ -8,69 +8,94 @@ import {Ownable2Step} from "oz/access/Ownable2Step.sol";
...
 
   event OrderCreated(address indexed order);
 
   /// @dev Allows to create a new proxy contract and execute a message call to the new proxy within one transaction.
   /// @param _singleton Address of singleton contract. Must be deployed at the time of execution.
   /// @param _initializer Payload for a message call to be sent to a new proxy contract.
-  /// @param saltNonce Nonce that will be used to generate the salt to calculate the address of the new proxy contract.
-  function createProxy(address _singleton, bytes memory _initializer, uint256 saltNonce)
+  /// @param _saltNonce Nonce that will be used to generate the salt to calculate the address of the new proxy contract.
+  function createProxy(address _singleton, bytes memory _initializer, uint256 _saltNonce)
```
Comment (NelsonGaldeman) [inline packages/contracts/src/OrderFactory.sol:22]: Good call here!

File: packages/contracts/src/OrderFactory.sol [hunk 3]
```diff
@@ -8,69 +8,94 @@ import {Ownable2Step} from "oz/access/Ownable2Step.sol";
...
     emit OrderCreated(order);
 
-    // Transfer the principal to the order
-    IERC20(_sellToken).transferFrom(msg.sender, order, _principal - ((_principal * protocolFee) / 100));
+    // Transfer the total amount to the order
+    IERC20(_sellToken).transferFrom(msg.sender, order, totalAmountWithoutFees);
 
     // Transfer the fee to the factory
-    IERC20(_sellToken).transferFrom(msg.sender, address(this), (_principal * protocolFee) / 100);
+    IERC20(_sellToken).transferFrom(msg.sender, address(this), (_totalAmount * protocolFee) / 100);
```
Comment (NelsonGaldeman) [inline packages/contracts/src/OrderFactory.sol]: I'm not entirely sure weather it's best (gas-wise) to re-do the math here or to store `feeAmount` and substract from the total (which is already in storage) every time. Substraction should be cheaper than multiplication & division. Can you do some profiling and update accordingly?

File: packages/contracts/src/OrderFactory.sol [hunk 4]
```diff
@@ -8,69 +8,94 @@ import {Ownable2Step} from "oz/access/Ownable2Step.sol";
...
+  /// @param _receiver The receiver of the buyToken orders.
+  /// @param _sellToken The token that is being traded in the order.
+  /// @param _totalAmount The total amount of the DCA order.
+  /// @param _buyToken The token that is DCA'd in the order.
+  /// @param _startTime The start time of the DCA order.
+  /// @param _endTime The end time of the DCA order.
+  /// @param _interval The frequency interval of the DCA order in hours.
+  /// @param _settlementContract The settlement contract address.
+  /// @param _saltNonce Nonce that will be used to generate the salt to calculate the address of the new order contract.
+  function createOrderWithNonce(
```
Comment (NelsonGaldeman) [inline packages/contracts/src/OrderFactory.sol:50]: Can you update the SDK/frontend to reflect contract changes in this same PR? We should re-deploy the contract before this PR is merged so it won't break

**Review by NelsonGaldeman — CHANGES_REQUESTED** [review]
Good PR, core changes are good. Just a few small comments here and there but good job man!

File: packages/contracts/src/OrderFactory.sol (hunk 4 above)
Comment (berteotti) [reply packages/contracts/src/OrderFactory.sol:50]: Will do it!

File: packages/contracts/src/OrderFactory.sol (hunk 3 above)
Comment (berteotti) [reply packages/contracts/src/OrderFactory.sol]: Did the math and it saves gas doing saving the `feeAmount`. Good tip!

Comment (berteotti) [reply packages/contracts/src/OrderFactory.sol]: Fixed

File: packages/contracts/src/DCAOrder.sol (hunk 1 above)
Comment (berteotti) [reply packages/contracts/src/DCAOrder.sol]: Fixed

**Review by NelsonGaldeman — APPROVED** [review]
LGTM

**Pull request merged**